File: examples/docs-code-snippets/src/main/java/examples/java/application/RegisterEventListener.java
package examples.java.application;

import com.ing.baker.runtime.javadsl.Baker;
import com.ing.baker.runtime.javadsl.EventInstance;

public class RegisterEventListener {

    private final Baker baker;

    public RegisterEventListener(Baker baker) {
        this.baker = baker;
    }

    public void example() {
        baker.registerEventListener((String recipeInstanceId, String event) ->
            System.out.println("Recipe Instance " + recipeInstanceId + " processed event " + event)
        );
    }
}

File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/application/RegisterEventListener.kt
package examples.kotlin.application

import com.ing.baker.runtime.kotlindsl.Baker

class RegisterEventListener(private val baker: Baker) {

    suspend fun example() {
        baker.registerEventListener { recipeInstanceId, eventInstance ->
            println("Recipe Instance: $recipeInstanceId, processed event: $eventInstance ")
        }
    }
}